src/wwwroot/app/services/ThesaurusService.js

```diff
@@ -16,41 +14,31 @@ const curryLength = 3;
 
 import THESAURUS_STRUCTURES from './ThesaurusStructuresStore.js';
 
-let _HttpService, _$q, _$translate;
+let _HttpService, _$q, _$translate, _HttpCacheService;
 let _curryAction = curry(_actionOfAdditionFieldsForTopic, curryLength);
 
-const cache = {};
-
 export default class ThesaurusService {
-   constructor(HttpService, $q, $translate) {
+   constructor(HttpService, $q, $translate, HttpCacheService) {
       'ngInject';
       _HttpService = HttpService;
       _$q = $q;
       _$translate = $translate;
+      _HttpCacheService = HttpCacheService;
    }
 
    getThesaurusTopics(thesaurusName) {
-      if (has(cache, thesaurusName)) {
-         return _$q.when(cache[thesaurusName]);
-      } else {
-         let thesaurusesToLoad = _getLoadedThesaurusesList(thesaurusName);
-         let mapThesaurusPromises = utils.array2map(thesaurusesToLoad, name => _HttpService.get(name));
-         return _$q.all(mapThesaurusPromises).then(thesauruses => {
-            forEach(thesauruses, (thesaurus, name) => {
-               cache[name] = thesaurus;
-               _actionOfAdditionFieldsForTopics(thesaurus, name, _addRefTextFieldFunction);
-            });
-            return cache[thesaurusName];
+      let thesaurusesToLoad = _getLoadedThesaurusesList(thesaurusName);
+      let mapThesaurusPromises = utils.array2map(thesaurusesToLoad, name => _HttpCacheService.get(name));
+      return _$q.all(mapThesaurusPromises).then(thesauruses => {
+         forEach(thesauruses, (thesaurus, name) => {
+            _actionOfAdditionFieldsForTopics(thesaurus, name, _addRefTextFieldFunction);
          });
-      }
+         return thesauruses[thesaurusName];
+      });
    }
 
    getThesaurusTopic(thesaurusName, id) {
-      if (has(cache, thesaurusName)) {
-         return _$q.when(find(cache[thesaurusName], {id}));
-      } else {
-         return this.getThesaurusTopics(thesaurusName).then(() => find(cache[thesaurusName], {id}));
-      }
+      return this.getThesaurusTopics(thesaurusName).then((topics) => find(topics, {id}));
```

Comment:
u can try to use `find` from `lodash/fp`
and your code can be like this one 
`return this.getThesaurusTopics(thesaurusName).then(find({id})`